project-ideas/fulfillment-center-mgmt/readme.md

```diff
@@ -143,9 +143,100 @@ The TO is an OrderType like SalesOrder and PurchaseOrder.
+
+1. Order Header - Transfer Order
+   1. Created
+   2. Approved
+   3. Completed
+   4. Cancelled
+   
+2. Order Item - Transfer Order Item
+   1. Created
+   2. Pending Fulfillment - newly introduced
+   3. Pending Receipt - newly introduced
+   4. Completed
+   5. Cancelled
+
+#### Transfer Order Life cycle
+
+1. TOs where Fulfillment location is managed by OMS and Receiving location is managed by third party e.g. Store to Warehouse
+   1. statusFlowId="TO_PendingFulfill"
+   2. Flow
+      1. TO Created in OMS
+         1. TO and TO Items in CREATED status
+      2. TO Approval
+         1. TO updated from status CREATED to APPROVED
+         2. TO Item: statusId="ITEM_CREATED" toStatusId="ITEM_PENDING_FULFILLMENT" transitionName="Approve & Pending Fulfillment Item"
+         3. Here item reservations should happen in approval since fulfillment in OMS.
+      3. TO Fulfillment
+         1. TO updated from status APPROVED to COMPLETED
+         2. TO Item: statusId="ITEM_PENDING_FULFILLMENT" toStatusId="ITEM_COMPLETED" transitionName="Complete Item" 
+      
+
+2. TOs where Fulfillment location is managed by third party and Receiving location is managed by OMS e.g. Warehouse to Store
+   1. statusFlowId="TO_PendingReceive"
+   2. Flow
+       1. TO Created in OMS
+           1. TO and TO Items in CREATED status
+       2. TO Approval
+           1. TO updated from status CREATED to APPROVED
+           2. TO Item: statusId="ITEM_CREATED" toStatusId="ITEM_PENDING_RECEIPT" transitionName="Approve & Pending Receipt Item"
+           3. Here No item reservations should happen in approval since only receiving in OMS.
+       3. TO Receiving
+           1. TO updated from status APPROVED to COMPLETED
+           2. TO Item: statusId="ITEM_PENDING_RECEIPT" toStatusId="ITEM_COMPLETED" transitionName="Complete Item"
+
+
+3. TOs where both Fulfillment and Receiving locations are managed by OMS e.g. Store to Store
+   1. statusFlowId="TO_PendingFulfillAndReceive"
+   2. Flow
+       1. TO Created in OMS
+           1. TO and TO Items in CREATED status
+       2. TO Approval
+           1. TO updated from status CREATED to APPROVED
+           2. TO Item: statusId="ITEM_CREATED" toStatusId="ITEM_PENDING_FULFILLMENT" transitionName="Approve & Pending Fulfillment Item"
+           3. Here item reservations should happen in approval since fulfillment in OMS.
+       3. TO Fulfillment
+          1. TO should still be in APPROVED status
+          2. TO Item: statusId="ITEM_PENDING_FULFILLMENT" toStatusId="ITEM_PENDING_RECEIPT" transitionName="Pending Receipt Item" 
+      4. TO Receiving
+            1. TO updated from status APPROVED to COMPLETED
+            2. TO Item: statusId="ITEM_PENDING_RECEIPT" toStatusId="ITEM_COMPLETED" transitionName="Complete Item"
+
+**NOTE**
+1. The Fulfillment App will list all TOs where Order is Approved and Items are Pending Fulfillment so that they are eligible to be shipped in OMS.
```

Comment:
As per discussion with @dixitdeepak Sir, the recommended way is to check on OrderHeader status and statusFlowId to find and list items for eligible TOs in the fulfillment app. Adding check on item status is not the query will be complex. Once the eligible orders are fetched, further eligible items can be fetched on the basis of Pending FUlfillment item status.
Similar logic to be used in Receiving app.

Reply:
this section needs update based on above comment, will update in upcoming commits

Reply:
Further discussed with @dt2patel and @dixitdeepak Sir, the current design holds good. We will use Solr for FindOrders, so this should not be a problem.